pyomo/core/base/indexed_component.py

```diff
@@ -40,6 +40,9 @@ def normalize_index(index):
     return idx
 normalize_index.flatten = True
 
+class _NotValid(object):
+    pass
```

Comment:
Having written this, I am wondering if this flag should be renamed `_NotPresent` (or `_NotFound`)?  I worry that it will become confusing having `param._NotValid`, which indicated that an invalid value is present (either as the Default or as the value of a scalar) along with `indexed_component._NotValid`.